lib/statsLogger.js

```diff
@@ -151,6 +163,18 @@ exports.createInstance = function(flushInterval, backend, backendConfig, stats)
   StatsLogger.prototype.flush = function() {
     var self = this;
 
+    if (self.captureFunctions.length > 0) {
+      for (var i = 0; i < self.captureFunctions.length; i++) {
+        var func = self.captureFunctions[i];
+        try {
+          self.recordStat(func.name, func.statFunction());
+        }
+        catch (e) {
+          console.log("Error collecting stat on flush:" + e);
```

Comment:
Instead of logging, this should probably follow the convention set in the rest of this code base, and emit the error. Then, you could also test that the error is being emitted properly.